elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -29,50 +123,37 @@ defmodule Web.Settings.DNS do
     </p>
     <section class="bg-white dark:bg-gray-900">
       <div class="max-w-2xl px-4 py-8 mx-auto lg:py-16">
+        <.flash kind={:success} flash={@flash} phx-click="lv:clear-flash" />
         <h2 class="mb-4 text-xl font-bold text-gray-900 dark:text-white">Client DNS</h2>
-        <form action="#">
+        <.form for={@form} phx-change={:change} phx-submit={:submit}>
           <div class="grid gap-4 mb-4 sm:grid-cols-1 sm:gap-6 sm:mb-6">
             <div>
-              <label
-                for="resolver"
-                class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
-              >
-                Resolver
-              </label>
-              <select
-                id="resolver"
-                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
-              >
-                <option>System default</option>
-                <option selected>Custom</option>
-              </select>
+              <.input
+                label="Resolver"
+                type="select"
+                field={@form[:resolver]}
+                options={@resolver_opts}
+                required
+              />
             </div>
+
             <div>
-              <.label for="resolver-address">
-                Address
-              </.label>
-              <input
-                type="text"
-                name="address"
-                id="resolver-address"
-                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500"
-                value="https://doh.familyshield.opendns.com/dns-query"
-                required
+              <.input
+                label="Address"
+                field={@form[:clients_upstream_dns]}
+                value={pretty_print_addrs(@form[:clients_upstream_dns].value)}
+                placeholder="DNS Server Address"
```

Comment:
```suggestion
                placeholder="DNS Server Address(es)"
```

Reply:
This is no longer needed as only one address is needed per input now.